### PR TITLE
Separate leftwm bin

### LINKED ISSUE #1190: Separate leftwm(bin) target from the leftwm crate
## What and how

I think we could separate the leftwm(bin) binary target from the leftwm crate. It itself only starts child processes and depends on about two functions from the util mod, one of them bein exclusively used by leftwm(bin) and the other being a simple `Vec::retain` shortcut for the `Vec<Child>` wrapper.

## Why

This should improve compile times, especially on multi-threading hardware, but also because it's one binary less to compile for the usual changes (6 less when changing only leftwm(bin)).

Anyone against this development change? I'd open a pr in a few days otherwise.

### PR DESCRIPTION
**Pull request by Eskaan**
# Description

Separate the leftwm(bin) target out of the leftwm crate into the leftwm-watchdog crate to improve compile times.

I changed the binary target's name to `leftwm` to avoid problems (this is now a development change, the target directory's executable are the same as before).

Fixes #1190 

## Type of change

- [x] Development change (no change visible to user)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation only update (no change to the factual codebase)
- [ ] This change requires a documentation update

# Checklist:

- [x] Ran `make test-full` locally with no errors or warnings reported
  Note: To fully reproduce CI checks, you will need to run `make test-full-nix`. Usually, this is not necessary.

### CODE REVIEW

**Review by mautamu — APPROVED** [review]
Thank you for this, Eskaan! Looks good.

File: leftwm-watchdog/src/utils.rs [hunk 1]
```diff
@@ -0,0 +1,315 @@
...
+
+pub const fn get_help_template() -> &'static str {
+    "\
+{name} {version}
+{author-with-newline}{about-with-newline}
+{usage-heading} {usage}
+
+{all-args}
+"
+}
```
Comment (Eskaan) [inline leftwm-watchdog/src/utils.rs:19]: I forgot to mention, the leftwm(bin) wasn't completely independent from the utils. These two functions should be the only duplicated code. I think 15 lines is worth the trade-off of (presumably) better compile times.

File: leftwm-watchdog/src/utils.rs [hunk 2]
```diff
@@ -0,0 +1,315 @@
...
+        value.to_lowercase().parse::<bool>().ok()
+    }
+}
+
+/// Register the `SIGCHLD` signal handler. Once the signal is received,
+/// the flag will be set true. User needs to manually clear the flag.
+pub fn register_child_hook(flag: Arc<AtomicBool>) {
+    _ = signal_hook::flag::register(signal_hook::consts::signal::SIGCHLD, flag)
+        .map_err(|err| println!("Cannot register SIGCHLD signal handler: {err:?}"));
+}
```
Comment (Eskaan) [inline leftwm-watchdog/src/utils.rs:221]: Second function.

**Comment by mautamu** [comment]
Performed timings estimate using `cargo build --timings`
- Initial build (from clean target repo) on eaa13b37d433ff7b9c4c5ad4fe2afc2e1bce2480: 50.0s
- Incremental build (modified leftwm/src/utils.rs trivially) on eaa13b37d433ff7b9c4c5ad4fe2afc2e1bce2480: 15.0s
- Initial build (from clean target repo) on 
cffad845b689d9dee1dc4877343b8eba6a: 43.9s
- Incremental build (modified leftwm/src/utils.rs trivially) on 71a115cffad845b689d9dee1dc4877343b8eba6a: 17.0s

Attached reports:
[cargo-timings.gz](https://github.com/leftwm/leftwm/files/13691719/cargo-timings.gz).

I think some duplicate code is worth the tradeoff for initial build time so will go ahead and merge this. Thanks again as always @Eskaan!

**Pull request merged**